src/inline-loading/inline-loading.component.ts

```diff
@@ -14,60 +27,88 @@ import {
 @Component({
 	selector: "ibm-inline-loading",
 	template: `
-		<div class="bx--inline-loading__animation">
+		<div *ngIf="state !== InlineLoadingState.Hidden"
+			class="bx--inline-loading__animation">
 			<div
-				*ngIf="success === false"
+				*ngIf="state === InlineLoadingState.Inactive || state === InlineLoadingState.Active"
 				class="bx--loading bx--loading--small"
 				[ngClass]="{
-					'bx--loading--stop': !isActive
+					'bx--loading--stop': state === InlineLoadingState.Inactive
 				}">
 				<svg class="bx--loading__svg" viewBox="-75 -75 150 150">
 					<circle class="bx--loading__background" cx="0" cy="0" r="30" />
 					<circle class="bx--loading__stroke" cx="0" cy="0" r="30" />
 				</svg>
 			</div>
 			<svg
-				*ngIf="success === true"
+				*ngIf="state === InlineLoadingState.Finished"
 				class="bx--inline-loading__checkmark-container bx--inline-loading__svg"
 				xmlns="http://www.w3.org/2000/svg"
 				viewBox="0 0 10 10">
 				<polyline class="bx--inline-loading__checkmark" points="0.74 3.4 3.67 6.34 9.24 0.74"></polyline>
 			</svg>
+			<svg
+				*ngIf="state === InlineLoadingState.Error"
+				class="bx--inline-loading--error"
+				xmlns="http://www.w3.org/2000/svg"
+				viewBox="0 0 16 16">
+				<path d="M8,1C4.1,1,1,4.1,1,8s3.1,7,7,7s7-3.1,7-7S11.9,1,8,1z M10.7,11.5L4.5,5.3l0.8-0.8l6.2,6.2L10.7,11.5z"></path>
+				<path fill="none" d="M10.7,11.5L4.5,5.3l0.8-0.8l6.2,6.2L10.7,11.5z" data-icon-path="inner-path" opacity="0"></path>
+			</svg>
+
 		</div>
-		<p *ngIf="success === false" class="bx--inline-loading__text">{{loadingText}}</p>
-		<p *ngIf="success === true" class="bx--inline-loading__text">{{successText}}</p>
+		<p
+			*ngIf="state === InlineLoadingState.Inactive || state === InlineLoadingState.Active"
+			class="bx--inline-loading__text">{{loadingText}}</p>
+		<p *ngIf="state === InlineLoadingState.Finished" class="bx--inline-loading__text">{{successText}}</p>
+		<p *ngIf="state === InlineLoadingState.Error" class="bx--inline-loading__text">{{errorText}}</p>
 	`
 })
 export class InlineLoading {
+	InlineLoadingState = InlineLoadingState;
+
 	/**
 	 * Specify the text description for the loading state.
 	 */
-	@Input() loadingText;
+	@Input() state: InlineLoadingState = InlineLoadingState.Active;
```

Comment:
I'm good with this as long as `[state]="'active'"` doesn't cause Angular to complain when it typechecks the templates 👍

Reply:
I haven't tested that scenario but why would somebody use the `string` literal instead of the `enum`?

Reply:
In the case that they don't bind the `enum` to the component class ... It's an extra step that can trip people up, so we've generally avoided it in favour of string type unions